components/dataflow/predict/src/predict.py

```diff
@@ -60,6 +61,14 @@ def parse_arguments():
                       type=int,
                       default=32,
                       help='Batch size used in prediction.')
+  parser.add_argument('--prediction-results-uri-pattern-output-path',
```

Comment:
I probably missed some context here. Why is this needed? Does it add burden to component author to add each output as a command line argument?

Reply:
The industry is moving away from Docker towards open and standardized container support and so do our users and customers.
We have received bug reports from users that our pipelines fail on non-Docker Kubernetes clusters.
While making Argo work on a non-Docker cluster is trivial, it's much harder with Pipelines.
One particular problem is with local artifact paths. There are specific system-imposed requirements for where the artifacts can and cannot be stored. More so, this depends on the Argo executor being used in the cluster.
De-hardcoding the local artifact paths seems to be the only option for keeping the pipelines portable.
Usually the programs are already written without any paths hard-coded.
If the program did not follow the best practices, the author only need to add one line to the program to resolve the problem. This seem to be a small price where the alternative is a component that only work on some clusters.